### PR TITLE
drivers: i2c: Move "Use 10-bit addressing" option to i2c_msg

### LINKED ISSUE #3806: I2C: move "Use 10-bit addressing" option to i2c_msg
**_Reported by Piotr Mienkowski:_**

The current support for 10-bit addressing in i2c.h file is incorrect: the "Use 10-bit addressing" configuration option defined as I2C_ADDR_10_BITS belongs to dev_config structure and is a parameter of i2c_configure() function. We treat it the same as speed, master/slave mode. i.e. it is a property which applies to all transactions performed on the bus. However, the I2C specification states: "Devices with 7-bit and 10-bit addresses can be connected to the same I2C-bus, and both 7-bit and 10-bit addressing can be used in all bus speed modes."
As such I2C_ADDR_10_BITS should be a property of i2c_msg.flags. In that case every message sent on the I2C bus can have independently set addressing mode.

(Imported from Jira ZEP-2373)

### PR DESCRIPTION
**Pull request by Mierunski**
Fixes #3806
Added new message flag I2C_MSG_ADDR_10_BITS while keeping the old flag for 10 bit addressing for backward compatibility.
Modified nrf i2c driver to check this flag instead, now all other drivers should be modified in the same way.

### CODE REVIEW

**Review by jakub-uC — APPROVED** [review]

File: drivers/i2c/i2c_nrfx_twi.c [hunk 1]
```diff
@@ -90,10 +94,6 @@ static int i2c_nrfx_twi_configure(struct device *dev, u32_t dev_config)
 {
 	nrfx_twi_t const *inst = &(get_dev_config(dev)->twi);
 
-	if (I2C_ADDR_10_BITS & dev_config) {
-		return -EINVAL;
```
Comment (anangl) [inline drivers/i2c/i2c_nrfx_twi.c]: I think this should stay. As long as the flag is available in the API, though it is marked as deprecated, it should be handled correctly to keep the backward compatibility.

**Review by anangl — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_nrfx_twi.c (hunk 1 above)
Comment (Mierunski) [reply drivers/i2c/i2c_nrfx_twi.c]: Added it back

**Comment by Mierunski** [comment]
@carlescufi @nashif Could we get this merged?

File: include/i2c.h [hunk 2]
```diff
@@ -52,7 +52,7 @@ extern "C" {
 #define I2C_SPEED_GET(cfg) 		(((cfg) & I2C_SPEED_MASK) \
 						>> I2C_SPEED_SHIFT)
 
-/** Use 10-bit addressing. */
+/** Use 10-bit addressing. DEPRECATED - Use I2C_MSG_ADDR_10_BITS instead. */
 #define I2C_ADDR_10_BITS		(1 << 0)
```
Comment (anangl) [inline include/i2c.h]: `I2C_SLAVE_FLAGS_ADDR_10_BITS` references this macro in line 66 (I can't put the comment there). I guess `(1 << 0)` should be used there directly.

**Review by anangl — APPROVED** [review]

**Comment by carlescufi** [comment]
@MaureenHelm @galak can you take a look at this one?

**Review by carlescufi — APPROVED** [review]

**Review by superna9999 — APPROVED** [review]

**Review by galak — COMMENTED** [review]
What about all the other controllers?  I wondering if we should say if the controller supports the feature at all, and than handle the error checking there?

**Pull request merged**